shared/tests/utility/math/TestCoordinates.cpp

```diff
@@ -0,0 +1,1029 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <array>
+#include <catch.hpp>
+#include <cfloat>
+#include <cmath>
+#include <fmt/format.h>
+#include <iomanip>
+#include <utility>
+
+#include "utility/math/coordinates.h"
+
+/**
+ * Test data and functions to validate coordinates.cpp functions.
+ *
+ * Test value domain to stay between the values of TEST_DBL_MAX = 5.9667e+153, and TEST_DBL_MIN = 1.4853e-307
+ * This means the usual domain edge cases - DBL_MAX and DBL_MIN - have been left out or replaced. It has been determined
+ * that inputs approaching these edge cases will not be possible in normal operation. See coordinates.h comments for
+ * more details.
+ * @author Liam Craft
+ */
+
+static constexpr double ERROR_THRESHOLD = 1e-6;
+static constexpr double TEST_DBL_MAX    = 5.9667e+153;
+static constexpr double TEST_DBL_MIN    = 1.4853e-307;
+
+
+/** Cartesian test coords */
+static const std::array<Eigen::Vector3d, 200> cart_coords = {
+    Eigen::Vector3d(-0.492109181440401e+19, -0.080404385128274e+19, -1.673196359027018e+19),
```

Comment:
If you have large sets of data like this in tests, it's generally best practice to separate the "logic" from the "data".
Put these constant sets into other files and include them. That way the logic of the test is clear in one file without getting hidden with large amounts of data like this.

Reply:
Good call. We implemented these changes promptly.....